packages/api-v2/src/hooks/useCryptoConfig.ts

```diff
@@ -1,16 +1,30 @@
+import { useCallback } from 'react';
+import useAuthorize from './useAuthorize';
 import useAuthorizedQuery from '../useAuthorizedQuery';
 
+/** A custom hook to get the cryptocurrencies config information from `crypto_config` endpoint. */
 const useCryptoConfig = () => {
-    // 15 seconds, as crypto_config potentially contains dynamic data
-    // so caching is there only to prevent multiple requests in a short time
-    // generally it should be fresh data
-    return useAuthorizedQuery(
-        'crypto_config',
-        {},
-        {
-            staleTime: 15 * 1000,
-        }
+    const { data: authorizeData, isLoading: isAuthorizeLoading } = useAuthorize();
+
+    const { data, isLoading: isCryptConfigLoading, ...rest } = useAuthorizedQuery('crypto_config');
+
+    const cryptoConfig = authorizeData?.currency
+        ? data?.crypto_config?.currencies_config[authorizeData?.currency]
+        : null;
+
+    const getConfig = useCallback(
```

Comment:
for upcoming PRs/updates/refactors: would be cool to get rid of that weird pattern of "getConfig"

Reply:
(out of scope for this PR - just suggestion for future as I found this pattern to be unnecesarily confusing)

Reply:
hey @wojciech-deriv 
just to clarify your concern
the reason i added `getConfig` here is for convenience to get the config for particular currency.
and also we are returning the config for active account.
so
as i understand your proposal can be to return the whole response and let the developer get the proper config in the consumer itself, right?
smth like:
```
import { useCallback } from 'react';
import useAuthorize from './useAuthorize';
import useAuthorizedQuery from '../useAuthorizedQuery';

/** A custom hook to get the cryptocurrencies config information from `crypto_config` endpoint. */
const useCryptoConfig = () => {
    const { data, ...rest } = useAuthorizedQuery('crypto_config');

    return {
        /** Available cryptocurrencies config for authorized account*/
        data: cryptoConfig,
        ...rest,
    };
};

export default useCryptoConfig;
```

Reply:
yes, thats what I meant, 
hmm, I see the point - personally I would use something like
const { currencyConfig } 
= useCurrencyConfig(currency); to get the same convenience without returning function from hook, 

either way, its minor/optional,